module1/lessons/hashes.md

```diff
@@ -0,0 +1,150 @@
+# Hashes
+
+### Learning Goals
+- Identify the differences between Hashes and Arrays. 
+- Practice building, accessing, and updating Hashes.
+
+### Vocabulary
+- Hash
+- Symbol
+- Key
+- Value
+
+### Warm-up
+Say we have an array named `grocery_list` that looks like this:
+```ruby
+grocery_list = ["milk", "eggs", "eggs", "eggs", "eggs", "eggs", "eggs", "avocado", "avocado", "tortilla", "tortilla", "tortilla", "tortilla", "tortilla", "tortilla", "tortilla", "tortilla", "tortilla"]
+```
+
+Answer these questions with your group:
+1. What is problematic about this data being in this array?
+1. How would you prefer this data be stored?
+
+
+## Part 1: Student Exploration
+
+With our groups, we will be working through [this repository](https://github.com/turingschool-examples/mod-1-be-exercises/blob/main/ruby_exercises/data-types/collections/spec/hashes_spec.rb)
+
+If you have already completed these exercises, delete your work and let's do it again! Repetition is key to learning. 
+
+While working through these exercises, write down any important information you come across. 
+Take note of:
+  - What was challenging? How did you end up solving that challenge?
+  - Any keypoints.
+  - What questions came up?
+  - What questions do you still have?
+  - Anything else that seems important.
+
+Resources to use while working through the exercises:
+  - Your breakout room mates!
+  - Google (ex: `Ruby how to make a new hash with a default value`)
+  - [Ruby Docs](https://ruby-doc.org/core-2.7.0/Hash.html)
+  - [Intro To Hashes](introducing_hashes.md)
+ 
+
+## Part 2: Discussion
```

Comment:
I am really liking this section. Feels like a great way for the students to explore, practice solving, and document questions that they still weren't able to understand/solve for.